rmw_fastrtps_cpp/src/TypeSupport.cpp

```diff
@@ -202,8 +202,10 @@ bool TypeSupport::serializeROSmessage(eprosima::fastcdr::Cdr &ser, const rosidl_
             {
                 case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_BOOL:
                     {
-                        bool sb = false, &b = *(bool*)field;
-                        if(b) sb = true; 
+                        bool sb = false;
+                        // don't cast to bool here because if the bool is
+                        // uninitialized the random value can't be deserialized
+                        if(*(uint8_t*)field) sb = true;
```

Comment:
I believe you that this change fixes the issue, but for my own edification, I'm having a hard time understanding this change.

Why does casting this to a `uint8_t` rather than a `bool` make any difference? In both cases `sb` is the thing being serialized and it is always either `true` or `false` and is never uninitialized.

Which cast are you expecting to happen here? Maybe it would be clearer if you used `reinterpret_cast<bool>` or `reinterpret_cast<uint8_t>` it would be clearer what's happening. Since `field` is only being read here, I don't see how casting it to a different type changes anything other than the branch the `if` statement takes. I feel like I'm missing something obvious.